### ISSUE TITLE
Incorrect validation accuracy when size of validation set is not divisible by batch_size

### ISSUE THREAD

**Issue by yzgysjr** [tflearn/tflearn#75]
Hi,

Thank you so much for providing us such a useful framework.

I recently find a problem.

In https://github.com/tflearn/tflearn/blob/master/tflearn/helpers/trainer.py#L801, the function evaluating validation loss and accuracy, there might be a problem when size of validation set is not divisble by batch_size.

```{python}
def evaluate_flow(session, ops_to_evaluate, dataflow):
        if not isinstance(ops_to_evaluate, list):
            ops_to_evaluate = [ops_to_evaluate]
        tflearn.is_training(False, session)
        dataflow.reset()
        dataflow.start()
        res = [0. for i in ops_to_evaluate]
        feed_batch = dataflow.next()
        n_batches = len(dataflow.batches)
        while feed_batch:
            r = session.run(ops_to_evaluate, feed_batch)
            for i in range(len(r)):
                res[i] += r[i] / n_batches # BUG here
            feed_batch = dataflow.next()
        return res
```

Could I make a pull request to fix the problem? Thank you!

**Comment by aymericdamien** [comment]
Hi, thanks for your help!
I am not clear where is the issue here? Can you please provide an example?
```python
n_batches = len(dataflow.batches)
        while feed_batch:
            r = session.run(ops_to_evaluate, feed_batch)
            for i in range(len(r)):
                res[i] += r[i] / n_batches # BUG here
            feed_batch = dataflow.next()
        return res
```
dataflow.batches is a list of batches (a batch is a list of indexes from the whole dataset).
n_batches is the total number of batch in the dataset, so it is at least 1.

You can send a pull request, so it is easy to see changes you propose.

**Comment by yzgysjr** [comment]
Hi, thank you very much for your quick response!

For example, if size of my validation set is 515 and my batch size is 128, it will be split into 5 batches, of size 128, 128, 128, 128 and 3.  Consider a simple circumstance, If I get accuracy of 0.0, 0.0, 0.0, 0.0, 1.0 on each of mini batch, the code will produce an accuracy of (0 + 0 + 0 + 0 + 1) / 5 = 0.2. However, the correct accuracy should be 3 / 515 = 0.005825.

**Comment by aymericdamien** [comment]
Ah yeah, there is indeed an issue here. Please go ahead and send a pull request.
Thanks for pointing that out :)
